src/typechecker.erl

```diff
@@ -1568,11 +1568,20 @@ type_check_logic_op(Env, Op, P, Arg1, Arg2) ->
             throw({type_error, boolop, Op, P, Ty1});
         {true, Cs2} ->
             {Ty2, VB2, Cs3} = type_check_expr(Env#env{ venv = UnionVarBindsSecondArg(Env#env.venv,VB1 )}, Arg2),
-            case subtype(Ty2, {type, P, bool, []}, Env#env.tenv) of
+            % Allow any() in second argument for shortcut operators
+            SndArgTy = if Op == 'andalso'; Op == 'orelse' -> type(any);
```

Comment:
This could perhaps be cleaner by breaking the shortcut ops out of the `type_check_logic_op` function and creating a separate `type_check_shortcut_ops`. Just an idea.

Reply:
Yes, I did that for `type_check_logic_op_in`. Might save that for future cleanups though.